e2e/tests-dfx/install.bash

```diff
@@ -81,3 +81,40 @@ teardown() {
     dfx canister create --all
     assert_command_fail dfx canister install --all --wasm "${archive:?}/wallet/0.10.0/wallet.wasm"
 }
+
+@test "install runs post-install tasks" {
+    dfx_start
+    # shellcheck disable=SC2094
+    cat <<<"$(jq '.canisters.e2e_project.post_install="sh -c \"echo hello\""' dfx.json)" >dfx.json
+
+    assert_command dfx canister create --all
+    assert_command dfx build
+
+    assert_command dfx canister install --all
+    assert_match hello
+    
+    # shellcheck disable=SC2094
+    cat <<<"$(jq '.canisters.e2e_project.post_install=["sh -c \"echo hello\"", "sh -c \"return 1\""]' dfx.json)" >dfx.json
+    assert_command_fail dfx canister install --all --mode upgrade
+    assert_match hello
+}
+
+@test "post-install tasks receive environment variables" {
+    dfx_start
+    # shellcheck disable=SC2094
+    cat <<<"$(jq '.canisters.e2e_project.post_install="sh -c \"echo hello $CANISTER_ID\""' dfx.json)" >dfx.json
```

Comment:
This PR goes to some trouble to make sure it includes environment variables for dependencies.  Could we add tests that demonstrate that those environment variables are present when expected?